stations-injector/src/main/java/workshop/stations/StationsInjector.java

```diff
@@ -85,17 +85,11 @@ private void inject(RoutingContext ctx) {
             fut.complete();
           }, false, ar -> {}));
 
-        rxReadGunzippedTextResource("cff-stop-2016-02-29__.jsonl.gz")
-          .map(StationsInjector::toEntry)
-          .repeatWhen(notification -> notification.map(terminal -> {
```

Comment:
I would not leave that part (`repeatWhen`) to the lab attendees. It's too difficult.

We could split the flow in 3 parts:
1. Create an Observable of terminals by reading the gzipped file and mapping to entry (attendee)
2. Use the created observable to invoke repeat and store the result in the original variable (us)
3. Transform the observable to put each item in the `stations` cache and subscribe (attendee)

Reply:
Yeah, I'm going to leave the repeat bit out.

Reply:
I've reread your comment. I think it's better to just leave it out. If it reaches end, they can always invoke `/inject` again. That should work, and if it doesn't, redeploy and `/inject`. 

I created an issue the other day to have a way to cancel an existing inject if an invocation to `/inject` comes along and restart the injection. That would be a more solid solution.